### PR TITLE
extended JsonResult structure

### LINKED ISSUE #7817: feat: improve `forge script` `--json` output and handling of the `--silent` flag
### Component

Forge

### Describe the feature you would like

Currently when passing the `--json` flag the following is returned:

https://github.com/foundry-rs/foundry/blob/f0d9eeced8ec01045b2849ea2cc3c72773282d70/crates/script/src/lib.rs#L496-L501

It would be beneficial to return more information, as defined by `ScriptResult`

https://github.com/foundry-rs/foundry/blob/f0d9eeced8ec01045b2849ea2cc3c72773282d70/crates/script/src/lib.rs#L462-L474

In turn it would also be useful to define a JSON schema for the output (https://github.com/foundry-rs/foundry/issues/7813)

I've also noticed that alongside the JSON a lot of other lines are written to stdout making it difficult to cleanly capture the output. If `--silent` is passed and the `--json` flag is enabled only the JSON should be returned. It appears `--silent` is currently only applied to the compilation output during the build phase.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by francesco-gaglione**
## Motivation
Fixes #7817 

## Solution
Extendend `JsonResult` struct

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgtm, pending @zerosnacks

**Comment by DaniPopes** [comment]
Can't we just reuse `result` directly which also implements serde?

**Comment by francesco-gaglione** [comment]
> Can't we just reuse `result` directly which also implements serde?

Can you explain it better? Sorry, first contribution here, I'm still studying this project

**Comment by francesco-gaglione** [comment]
Other quesiton, should this change must be stated in the documentation?

File: crates/script/src/lib.rs [hunk 1]
```diff
@@ -492,9 +492,15 @@ impl ScriptResult {
...
 #[derive(Serialize, Deserialize)]
 struct JsonResult {
+    success: bool,
+    labeled_addresses: HashMap<Address, String>,
     logs: Vec<String>,
+    traces: Traces,
     gas_used: u64,
+    address: Option<Address>,
     returns: HashMap<String, NestedValue>,
+    breakpoints: Breakpoints,
```
Comment (zerosnacks) [inline crates/script/src/lib.rs:502]: I don't think it is necessary to include `breakpoints` here

**Review by zerosnacks — CHANGES_REQUESTED** [review]
Thanks for your PR! One small note

I don't think it is necessary to document this extensively given that the changes are additive and non-breaking. Users should be able to rely on the existence of certain keys but not that the entire JSON object remains as is or its size is consistent.

**Comment by DaniPopes** [comment]
> > Can't we just reuse `result` directly which also implements serde?
> 
> Can you explain it better? Sorry, first contribution here, I'm still studying this project.
> 
> Sorry for editing... I checked now understand...
> 
> Probably we can't for this reason:
> 
> `returns: self.execution_artifacts.returns.clone(),`
> 
> that is not included in `result`, so if this is important I think we cannot use `result`

You can keep the existing fields, and just add a `result` field

**Comment by francesco-gaglione** [comment]
@DaniPopes  `TestResult` is out of the crate and importing the forge crate conduct to a circular dependency. So I think we cannot use `TestResult`

**Review by zerosnacks — APPROVED** [review]

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**